zio-schema-json/shared/src/main/scala/zio/schema/codec/JsonCodec.scala

```diff
@@ -23,7 +23,7 @@ object JsonCodec extends Codec {
     )
 
   override def decoder[A](schema: Schema[A]): ZPipeline[Any, String, Byte, A] =
-    ZPipeline.utfDecode >>> ZPipeline.mapZIO(
+    ZPipeline.fromChannel(ZPipeline.utfDecode.channel.mapError(_.toString)) >>> ZPipeline.mapZIO(
```

Comment:
@adamgfraser I tried that, but it doesn't seem to work. I get `Unexpected end of input` in the `JsonCodecSpec` tests. Any suggestions?

Reply:
`utfDecode` tries to determine the encoding by reading a BOM from the beginning of the stream, and defaults to `UTF-8` if it could not find any.

I will check if there is something wrong with this logic but here you should be able to just use `utf8Decode` as JSON should always be in `UTF-8`.

Reply:
..but that does not solve the issue, so I'm looking into it more

Reply:
I think you ran into this issue: https://github.com/zio/zio-schema/issues/182 because the new `utfDecode` rechunks the stream

Reply:
@vigoo Thanks for your suggestions and thanks for adding `ZPipeline.mapError`. I still have the same error (`Unexpected end of input`) using `utf8Decode`. Will check https://github.com/zio/zio-schema/issues/182